### PR TITLE
docs: updates of ctags(1) and docs/ updates

### PR DESCRIPTION
**Pull request by hirooih**
- ctags(1) updates
  - markups fixes
    - 5c473165 docs: ctags(1): more consistent markups
    - 0da01779 docs: ctags(1): not use Sphinx extensions
    - 04e8a42c docs: ctags(1): additional markup fixes
  - contents reordering
    - e0fd0467 docs: ctags(1): reorder tag entries related descriptions
  - contents fixes (**please review carefully**)
    - 92a648ca docs: ctags(1): misc fixes
    - 2f6f62bd docs: ctags(1): update TAGS FILE FORMAT and TAG ENTRIES
    - dba2d592 docs: ctags(1): update OPTIONS
- docs fixes
  - db6aec7a docs: update and deal with TODO list

### CODE REVIEW

File: docs/man/ctags.1.rst
Comment (masatake) [inline docs/man/ctags.1.rst:247]: Thank you very much. This is an important suggestion. How do you think about introducing `.. TODO(code):` markup for the suggestions for improving the program code, not for the documents? So I can pick up this.

**Comment by masatake** [comment]
Could you split the pull request into two?
```

    5c47316 docs: ctags(1): more consistent markups
    0da0177 docs: ctags(1): not use Sphinx extensions
    04e8a42 docs: ctags(1): additional markup fixes
```

I would like you to merge this after updating man/README.
Could you consider and fix the markup rule for the man pages and write it to man/README.
We, the ctags developers recognize you as the maintainer of man/README.
The developers must get an ack from you when changing man/README.

The README is needed to write consistent man pages. I add a man page in #2844.
I wonder whether the new man page satisfies the rules that will be written in the README.

Thanks in advance.

**Comment by hirooih** [comment]
@masatake san,

Thank you for your feedbacks.

> Could you split the pull request into two?

I also wanted to do that, but it is very very hard.

>    5c47316 docs: ctags(1): more consistent markups

This is the first commit in this pull-request.

> docs: ctags(1): reorder tag entries related descriptions …

This is the third commit.  I reordered the contents aggressively.  This made it very hard to split the pull request.

>    0da0177 docs: ctags(1): not use Sphinx extensions
>    04e8a42 docs: ctags(1): additional markup fixes

I admit that these last two commits are my omissions (TENUKI, in Japanese).
I've squashed 0da0177 into "2f6f62b docs: ctags(1): update TAGS FILE FORMAT and TAG ENTRIES", and 04e8a42  into "5c47316 docs: ctags(1): more consistent markups".

I hope you accept this.

***

I am already updating man/README.  I will send another pull-request soon.
After that I am happy to propose update of the help message.

***

> How do you think about introducing .. TODO(code): markup ...

It is good idea.  I've fixed the comment.
I will also add this rule in man/README.

***

> [jet@living]~/var/ctags-new/man% echo --./*
> echo --./*
> zsh: no matches found: --./*

First I thought this was a warning message caused by your personal setting for zsh.  But I found that it is an error, and was the 2nd significant incompatibility "features" of zsh.

[Z-Shell Frequently-Asked Questions, 2.1: Differences from sh and ksh](http://zsh.sourceforge.net/FAQ/zshfaq02.html#l10)

> The next most classic difference is that unmatched glob patterns cause the command to abort; set NO_NOMATCH for those.

zsh users should know about this feature.  Search "zsh: no matches found", then you can find many questions about this error.  I think ctags(1) man page should not take care of it.  When I made the ctags-faq(7), this is one of the candidates to add to it.

> Anyway, this change is nothing about 'not use Sphinx extensions '.

It has been in "docs: ctags(1): update OPTIONS", since I committed last night. I think it is a proper commit for this.

File: docs/man/ctags.1.rst
Comment (masatake) [inline docs/man/ctags.1.rst]:
```
If specifying nothing as the parameter ``flags``, all fields are disabled.
e.g. ``--fields-C=`` means disabling all language-specific fields of C.
``--fields-all=`` means disabling all language-specific fields of all languages.
```

**Comment by masatake** [comment]
Does TODO mean your TODO, our TODO, or a question to @mastake?

I want to answer your questions. So I would like you to use special markers for this purpose.
I guess, in some cases, you have questions, but you don't need answers for them now because you will not reflect the answers to this pull request.

So, how about using the following markers: `Q(now)`, and `Q`.

I will answer for `Q(now)` while reviewing this pull request.

`Q(now)` is just an idea. You can use a better marker for the questions that @mastake should answer now.

**Comment by masatake** [comment]
My review board
- IN PROGRESS
  - docs: ctags(1): update TAGS FILE FORMAT and TAG ENTRIES 
- LGTM
  - docs: update and deal with TODO list 
  - docs: ctags(1): update OPTIONS

**Comment by hirooih** [comment]
@masatake san,

Thank you for sharing the review status.  Please take your time.

> Does TODO mean a documentation TODO or a question to @mastake?

As you may have noticed, it is case by case.  I should clarify my intentions more. I will try that in my future pull-requests.

>	.. TODO: "to all languages;" instead of "to all fields;" ???

In this case this was a question to myself.  I found that the original description was correct and I removed the TODO comment in "docs: ctags(1): update OPTIONS".

Now I find that I was able to squash "docs: ctags(1): update OPTIONS" to "docs: ctags(1): misc fixes".  These commits are not affected by reordering in "docs: ctags(1): reorder tag entries related descriptions".

File: docs/man/ctags.1.rst
Comment (masatake) [inline docs/man/ctags.1.rst:1584]:
```
are written into the tag file for explaining the tag file itself.
They are started from `!_`.
````

File: docs/man/ctags.1.rst
Comment (masatake) [inline docs/man/ctags.1.rst:1748]: Nice!

File: docs/man/ctags.1.rst
Comment (masatake) [inline docs/man/ctags.1.rst:1973]: TEMPORARY: MY BOOKMARK

File: docs/man/ctags.1.rst
Comment (masatake) [inline docs/man/ctags.1.rst:812]: Could you keep ``--param-<LANG>`` as a comment?
Another contributor may think why there is no description for the option.
Keeping this as a comment may help the person understand the situation.

File: docs/man/ctags.1.rst
Comment (masatake) [reply docs/man/ctags.1.rst:1748]: It will be nice having similar examples for `scope:`.

Comment (masatake) [reply docs/man/ctags.1.rst:1748]: These are just comments. Reflecting on the comment to this pull request is not a must.

File: docs/man/ctags.1.rst
Comment (masatake) [inline docs/man/ctags.1.rst:1261]: I remember  I implemented an option for enabling/disabling roles:
```
--roles-<LANG>.kind=[+|-]role
```

File: docs/man/ctags.1.rst
Comment (masatake) [inline docs/man/ctags.1.rst:1434]: I think `The' is not needed here.

**Comment by masatake** [comment]
@hirooih, thank you very much. This is great work.
I wrote some minor comments.
Other than them, the changes look good to me.

File: docs/man/ctags.1.rst
Comment (hirooih) [reply docs/man/ctags.1.rst:1749]: > It will be nice having similar examples for scope:.

That is a good idea.  I will add it in another pull-request.

File: docs/man/ctags.1.rst
Comment (hirooih) [reply docs/man/ctags.1.rst]: I've removed the note of the restriction.

File: docs/man/ctags.1.rst
Comment (hirooih) [reply docs/man/ctags.1.rst]: Whoops!  FIxed.

**Comment by hirooih** [comment]
@masatake san,
Thank you for your review.

> @hirooih, thank you very much. This is great work.

You are so welcome.
I also thank you for your great work on this project.

I will work for the examples for scope in another pull-request.  It needs a new subsection.
I fixed other suggestions.

**Review by masatake — APPROVED** [review]

**Pull request merged**
